src/components/tasks/task-dialog.tsx

```diff
@@ -828,18 +833,29 @@ const TaskDialog = ({ projectId, milestoneId: milestoneIdFromProps, open, task,
       (selected as string[]).map((id) => (options as Record<string, string>)[id]).join(", ");
   };
 
+  /**
+   *  Checks if project status is planning or initiation
+   */
+  const hasProjectStarted = () => {
+    return project?.status !== ProjectStatus.Planning && project?.status !== ProjectStatus.Initiation;
+  };
+
   /**
    * Returns min date for startDate input
    */
   const getStartDateMin = () => {
-    return selectedMilestone ? getValidDateTimeOrThrow(selectedMilestone.startDate) : undefined;
+    if (!selectedMilestone || checkIfMilestonesAreFixed()) {
+      return undefined;
+    }
+
+    return getValidDateTimeOrThrow(selectedMilestone.startDate);
   };
 
   /**
    * Returns max date for startDate input
    */
   const getStartDateMax = () => {
-    if (!selectedMilestone) return taskData.endDate ?? undefined;
+    if (!selectedMilestone || checkIfMilestonesAreFixed()) return taskData.endDate ?? undefined;
```

Comment:
Is the checkIfMilestonesAreFixed still defined somewhere?

Reply:
The name was changed to hasProjectStarted, I've updated all the references now. Also the logic was flipped from the previous suggestions, so I updated usage in the conditions to be !hasProjectStarted.